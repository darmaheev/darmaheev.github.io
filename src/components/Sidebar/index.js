import React from "react";
import PropTypes from "prop-types";
import {
  Wrapper,
  Toggle,
  ToggleButton,
  Sidebar,
  Nav,
  NavItem
} from "./styles";

const SidebarComponent = ({
  showSidebar,
  handleShowSidebar,
  ...restProps
}) => (
  <Wrapper hiddenSidebar={!showSidebar}>
    {showSidebar && <Sidebar {...restProps}>
      <Nav>
        <NavItem activeClassName="active" to="/search">
        Search
        </NavItem>
        <a href="https://github.com/darmaheev/darmaheev.github.io">
        About
        </a>
      </Nav>
    </Sidebar>}
    <Toggle>
      <ToggleButton
        hiddenSidebar={!showSidebar}
        onClick={handleShowSidebar}
      />
    </Toggle>
  </Wrapper>
);

SidebarComponent.defaultProps = {
  showSidebar: false,
  handleShowSidebar: () => {}
};

SidebarComponent.propTypes = {
  showSidebar: PropTypes.bool,
  handleShowSidebar: PropTypes.func
};

export default SidebarComponent;
