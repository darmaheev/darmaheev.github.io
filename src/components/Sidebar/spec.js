import React from "react";
import Sidebar from "./index";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders sidebar", () => {
  const sidebar = shallow(<Sidebar />);
  expect(sidebar).toMatchSnapshot();
});
