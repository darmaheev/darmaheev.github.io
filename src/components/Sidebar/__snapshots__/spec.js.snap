// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sidebar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarComponent
    handleShowSidebar={[Function]}
    showSidebar={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        <ForwardRef>
          <ForwardRef
            hiddenSidebar={true}
            onClick={[Function]}
          />
        </ForwardRef>,
      ],
      "hiddenSidebar": true,
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            hiddenSidebar={true}
            onClick={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "hiddenSidebar": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": false,
              "rules": Array [
                "
  border: none;
  display: inline;
  padding: 0;
  margin: 0;
  background: url(",
                "toggle-icon.png",
                ");
  width: 20px;
  height: 13px;
  ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  background: rgba(0, 43, 67, 0.1);
  height: 100%;
  width: 50px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 44px 0;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  background: #ffffff;
  display: flex;
  flex-direction: row;
  flex-shrink: 0;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          <ForwardRef>
            <ForwardRef
              hiddenSidebar={true}
              onClick={[Function]}
            />
          </ForwardRef>,
        ],
        "hiddenSidebar": true,
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              hiddenSidebar={true}
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "hiddenSidebar": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": false,
                "rules": Array [
                  "
  border: none;
  display: inline;
  padding: 0;
  margin: 0;
  background: url(",
                  "toggle-icon.png",
                  ");
  width: 20px;
  height: 13px;
  ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.button",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  background: rgba(0, 43, 67, 0.1);
  height: 100%;
  width: 50px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 44px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "rules": Array [
            "
  background: #ffffff;
  display: flex;
  flex-direction: row;
  flex-shrink: 0;
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
